### ISSUE TITLE
TimeoutError: Missing credentials in config

### ISSUE THREAD

**Issue by barnull** [arabold/serverless-autoprune-plugin#1]
Getting the following error when trying to use this plugin:

```
> serverless function prune -a
Serverless: Pruning specified functions in the following regions: us-west-2
Serverless: | { TimeoutError: Missing credentials in config
    at ClientRequest.<anonymous> (/Users/barnull/code/foo/api/node_modules/aws-sdk/lib/http/node.js:56:34)
    at ClientRequest.g (events.js:286:16)
    at emitNone (events.js:86:13)
    at ClientRequest.emit (events.js:185:7)
    at Socket.emitTimeout (_http_client.js:593:10)
    at Socket.g (events.js:286:16)
    at emitNone (events.js:86:13)
    at Socket.emit (events.js:185:7)
    at Socket._onTimeout (net.js:333:8)
    at tryOnTimeout (timers.js:224:11)
    at Timer.listOnTimeout (timers.js:198:5)
  cause:
   { TimeoutError: Missing credentials in config
       at ClientRequest.<anonymous> (/Users/barnull/code/foo/api/node_modules/aws-sdk/lib/http/node.js:56:34)
       at ClientRequest.g (events.js:286:16)
       at emitNone (events.js:86:13)
       at ClientRequest.emit (events.js:185:7)
       at Socket.emitTimeout (_http_client.js:593:10)
       at Socket.g (events.js:286:16)
       at emitNone (events.js:86:13)
       at Socket.emi%
```

**Comment by arabold** [comment]
Do other commands (`function deploy -a` for example) work as expected? The plugin should pull the AWS credentials from the current environment. However, it's very well possible I have a bug and forgot to assume the proper IAM role or something. I'll take a look...

**Comment by arabold** [comment]
Okay, I found something. Would you mind testing the latest version 0.1.3 with your setup? You _might_ have to specify the stage (`--stage dev`) in order for Serverless to pick the right credentials.

**Comment by michalsanger** [comment]
This worked for me:
`serverless function prune --stage dev --region eu-central-1`

**Comment by barnull** [comment]
The latest version worked but I had to run `prune` twice because the first time it was rate limited before completing:

```
> serverless function prune --stage dev --region us-west-2
Serverless: Pruning specified functions in the following regions: us-west-2
Serverless: - /Users/barnull/venvs/foo/lib/node_modules/serverless/node_modules/bluebird/js/release/async.js:61
        fn = function () { throw arg; };
                           ^

TooManyRequestsException: Rate exceeded
    at Object.extractError (/Users/barnull/venvs/foo/lib/node_modules/serverless/node_modules/aws-sdk/lib/protocol/json.js:43:27)
    at Request.extractError (/Users/barnull/venvs/foo/lib/node_modules/serverless/node_modules/aws-sdk/lib/protocol/rest_json.js:37:8)
    at Request.callListeners (/Users/barnull/venvs/foo/lib/node_modules/serverless/node_modules/aws-sdk/lib/sequential_executor.js:105:20)
    at Request.emit (/Users/barnull/venvs/foo/lib/node_modules/serverless/node_modules/aws-sdk/lib/sequential_executor.js:77:10)
    at Request.emit (/Users/barnull/venvs/foo/lib/node_modules/serverless/node_modules/aws-sdk/lib/request.js:615:14)
    at Request.transition (/Users/barnull/venvs/foo/lib/node_modules/serverless/node_modules/aws-sdk/lib/request.js:22:10)
    at AcceptorStateMachine.runTo (/Users/barnull/venvs/foo/lib/node_modules/serverless/node_modules/aws-sdk/lib/state_machine.js:14:12)
```

**Comment by ajagnanan** [comment]
Running into rate limiting issue too 😢

**Comment by arabold** [comment]
The rate limiting error is caused by a bug in Serverless: https://github.com/serverless/serverless/pull/1025

It's fixed in `master` branch but pending the 0.5.6 release. According to the team this should come any day now.

**Comment by arabold** [comment]
I've released a new version of the Auto-Prune plugin:

### 0.2.0
- Finally the plugin deserves it's name: Auto-Pruning! Passin in `--prune` to your 
  `serverless function deploy` will automatically delete previous Lambda versions.
- Handle AWS Rate Limit responses using a temporary fix until Serverless 0.5.6 is released

Let me know if this fixes your rate limiting errors.
